src/modules/sub-items/sub.items.module.js

```diff
@@ -141,51 +199,54 @@ export default class SubItemsModule extends Component {
         }
 
         const locations = response.View.Result.searchHits.searchHit;
-        const promises = [];
-
-        promises.push(
-            new Promise((resolve) => {
-                const contentIds = locations.map((item) => item.value.Location.ContentInfo.Content._id);
-
-                loadContentInfo(restInfo, contentIds, resolve);
-            }).then((contentInfo) => ({
-                locations,
-                totalCount: response.View.Result.count,
-                content: contentInfo.View.Result.searchHits.searchHit,
-            }))
-        );
 
-        return Promise.all(promises);
+        return new Promise((resolve) => {
+            const contentIds = locations.map((item) => item.value.Location.ContentInfo.Content._id);
+
+            loadContentInfo(restInfo, contentIds, (contentInfo) => {
+                const { totalCount: oldTotalCount } = this.state;
+                const totalCount = response.View.Result.count;
+
+                if (oldTotalCount !== totalCount) {
+                    this.discardAllPagesOtherThan(page);
```

Comment:
This is strange approach for naming methods: `discardAllPagesOtherThan()`. I would rename it to `discardPages(<array-of-excluded-pages-from-discard>)`.
What is a result of discard action?